### PR TITLE
(chapter 3.5)Fixed the inconsistency issue in QFT

### PR DESCRIPTION
**Pull request by singhmeet11**
The issue resolved is given in #970 

[3.5] Fixed the inconsistency in QFT chapter's

# Changes made
 The main aim was to eradicate the confusion between the |x> in section 1 and |x> in section 4. I have made the following changes:

1. |x> was not defined clearly, I defined a |X>(so that people don't get confused again) , I also changed the respective index from i to j (for X) and from i to k(for Y). 
2.  The mapping from |x> to |y> (which seemed confusing to me, because it nowhere mentions that we are talking about the whole vector or just one element), I used |j> and |k> for making this clearer .
3. The last change was in the U_{QFT} operator where I changed |x> and |y> to |j> and |k>.

No changes have been made anywhere except section 1. 

 

# Justification

1. This resolves the issue of intermixing of notations in section 1 and section 4. As there is a clear distinction between |X> and |x>.
2. Further replacing x and y by j and k in the powers of e, at a few places makes it easier to interpret the equations.
3. The overall linear algebra fits much better

### CODE REVIEW

**Review by divshacker — CHANGES_REQUESTED** [review]
Hey @singhmeet11  you have just added two extra files which are not required. You have to just edit the quantum-fourier-transform.ipynb file and just make a PR which you have done already. but you need to delete these extra files you have added in your branch

**Comment by singhmeet11** [comment]
@divshacker thanks for helping me out. Does this look good ?? 
And sorry if its still wrong a little new to all this

**Review by divshacker — DISMISSED** [review]
Now it's looks good.

File: content/ch-algorithms/quantum-fourier-transform.ipynb [hunk 1]
```diff
@@ -55,7 +55,7 @@
     "\n",
     "where $\\omega_N^{jk} = e^{2\\pi i \\frac{jk}{N}}$.\n",
     "\n",
-    "Similarly, the quantum Fourier transform acts on a quantum state $\\sum_{i=0}^{N-1} x_i \\vert i \\rangle$ and maps it to the quantum state $\\sum_{i=0}^{N-1} y_i \\vert i \\rangle$ according to the formula\n",
+    "Similarly, the quantum Fourier transform acts on a quantum state $\\vert X\\rangle =\\vert x_0  x_1  ...  x_{N-1}\\rangle= \\sum_{i=0}^{N-1} x_j \\vert j \\rangle$ and maps it to the quantum state $\\vert Y\\rangle = \\vert y_0 y_1 ... y_{N-1}\\rangle = \\sum_{i=0}^{N-1} y_k \\vert k \\rangle$ according to the formula\n",
```
Comment (frankharkins) [inline content/ch-algorithms/quantum-fourier-transform.ipynb:58]:
```suggestion
    "Similarly, the quantum Fourier transform acts on a quantum state $\\vert X\\rangle = \\sum_{j=0}^{N-1} x_j \\vert j \\rangle$ and maps it to the quantum state $\\vert Y\\rangle = \\sum_{k=0}^{N-1} y_k \\vert k \\rangle$ according to the formula\n",
```

File: content/ch-algorithms/quantum-fourier-transform.ipynb [hunk 2]
```diff
@@ -66,14 +66,14 @@
     "This can also be expressed as the map:\n",
     "\n",
     "\n",
-    "$$\\vert x \\rangle \\mapsto \\frac{1}{\\sqrt{N}}\\sum_{y=0}^{N-1}\\omega_N^{xy} \\vert y \\rangle$$\n",
+    "$$\\vert j \\rangle \\mapsto \\frac{1}{\\sqrt{N}}\\sum_{y=0}^{N-1}\\omega_N^{jk} \\vert k \\rangle$$\n",
```
Comment (frankharkins) [inline content/ch-algorithms/quantum-fourier-transform.ipynb:69]:
```suggestion
    "$$\\vert j \\rangle \\mapsto \\frac{1}{\\sqrt{N}}\\sum_{k=0}^{N-1}\\omega_N^{jk} \\vert k \\rangle$$\n",
```

**Review by frankharkins — CHANGES_REQUESTED** [review]
Thanks for having a go at this! I've suggested some changes:
- I think the `\\vert x_0  x_1  ...  x_{N-1}\\rangle` is maybe misleading as we generally use the label of a ket to refer to it's computational basis state, whereas here it seems `x_i` is the amplitude of the ith computational basis state.
- There are a few places I think the letters don't match up, I've suggested changes that I think makes the equations look right.

these equations are very dense though so it's possible I'm mistaken. Let me know what you think. 

Thanks!

File: content/ch-algorithms/quantum-fourier-transform.ipynb (hunk 1 above)
Comment (frankharkins) [inline content/ch-algorithms/quantum-fourier-transform.ipynb:58]:
```suggestion
    "Similarly, the quantum Fourier transform acts on a quantum state $\\vert X\\rangle = \\sum_{j=0}^{N-1} x_j \\vert j \\rangle$ and maps it to the quantum state $\\vert Y\\rangle = \\sum_{k=0}^{N-1} y_k \\vert k \\rangle$ according to the formula\n",
```

**Review by frankharkins — APPROVED** [review]
Thanks!

**Pull request merged**